### DISCUSSION TITLE
Rails 6 or Rails 7

### DISCUSSION THREAD

**Discussion by igbanam** [RubyConf-NG/openemrs#33] [General]
Currently, the app is built on Rails 6. I am contemplating its move to Rails 7. Now this wouldn't be an upgrade, but an overhaul. What triggered this ask is our juggling Sprockets and Webpacker when scaffolding; while Rails 7 handles this cleaner with importmaps and embracing more recent JS tech. 

What's to lose? — We currently have Doctors and Patients as the two resources on here. And we are not close to MVP yet, so I'm thinking we don't have much to lose with the rewrite, and it is not as costly.

What do you think?

**Comment by robocopkaka** [comment]
We can move to Rails 7. Best to tackle it now before we add in more resources. I've tried migrating from Rails 5 to Rails 6 on a large project and it was time consuming

**Comment by robocopkaka** [comment]
I'm going to create a new branch to work on it

**Reply by robocopkaka** [reply]
I should also bump ruby to  3.0 from 2.7.2

**Comment by robocopkaka** [comment]
https://github.com/RubyConf-NG/openemrs/pull/34

Ruby upgrade step 1

**Comment by igbanam** [comment]
I'm even suggesting we nuke this one and restart the project in Rails 7 with Tailwind and Postgres (or MySQL, I don't have a preference at this stage). The entity-space is small enough that we can do this.

**Reply by robocopkaka** [reply]
Oh. That makes sense. Do you want to take this up or I could start on it later tonight

**Reply by igbanam** [reply]
I see you already started in https://github.com/RubyConf-NG/openemrs/tree/chore/migrate-to-rails-7

**Comment by igbanam** [comment]
Some resources for the upgrade…

- [How to upgrade Rails 6.1 to Rails 7 ](https://dev.to/thomasvanholder/how-to-upgrade-rails-61-to-rails-7-33a3) is a sane guide
- [RailsDiff](https://railsdiff.org/6.1.0/7.0.4) will help us review properly
- The [Rails Guide](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) is always a good resource to have
